support-workers/src/main/scala/com/gu/zuora/ProductSubscriptionBuilders.scala

```diff
@@ -26,38 +32,38 @@ object ProductSubscriptionBuilders {
       case None => throw new CatalogDataNotFoundException(s"RatePlanId not found for $productDescription")
     }
 
-  implicit class ContributionSubscriptionBuilder(val contribution: Contribution) extends ProductSubscriptionBuilder {
-    def build(requestId: UUID, config: ZuoraConfig): SubscriptionData = {
-      val contributionConfig = config.contributionConfig(contribution.billingPeriod)
-      buildProductSubscription(
-        requestId,
-        contributionConfig.productRatePlanId,
-        List(
-          RatePlanChargeData(
-            ContributionRatePlanCharge(contributionConfig.productRatePlanChargeId, price = contribution.amount) //Pass the amount the user selected into Zuora
-          )
+  def buildContributionSubscription(contribution: Contribution, requestId: UUID, config: ZuoraConfig): SubscriptionData = {
+    val contributionConfig = config.contributionConfig(contribution.billingPeriod)
+    buildProductSubscription(
+      requestId,
+      contributionConfig.productRatePlanId,
+      List(
+        RatePlanChargeData(
+          ContributionRatePlanCharge(contributionConfig.productRatePlanChargeId, price = contribution.amount) //Pass the amount the user selected into Zuora
         )
       )
-    }
+    )
   }
 
-  implicit class DigitalPackSubscriptionBuilder(val digitalPack: DigitalPack) extends ProductSubscriptionBuilder {
-    def build(
+  object buildDigitalPackSubscription {
+    def apply(
+      digitalPack: DigitalPack,
       requestId: UUID,
-      config: ZuoraConfig,
       country: Country,
-      maybePromoCode: Option[PromoCode],
-      paymentMethod: Either[PaymentMethod, RedemptionData],
+      paymentMethod: Either[(ZuoraDigitalPackConfig, Option[PromoCode]), RedemptionData],
       promotionService: PromotionService,
       stage: Stage,
-      isTestUser: Boolean
-    ): Either[PromoError, SubscriptionData] = {
-
-      val contractEffectiveDate = LocalDate.now(DateTimeZone.UTC)
-      val contractAcceptanceDate = contractEffectiveDate
-        .plusDays(config.digitalPack.defaultFreeTrialPeriod)
-        .plusDays(config.digitalPack.paymentGracePeriod)
-
+      isTestUser: Boolean,
+      getCodeStatus: GetCodeStatus,
+      now: () => LocalDate
+    )(implicit ec: ExecutionContext): EitherT[Future, String, SubscriptionData] = {
+
+      val contractEffectiveDate = now()
+      val delay = paymentMethod match {
+        case Left((config, _)) => config.defaultFreeTrialPeriod + config.paymentGracePeriod
+        case Right(_) => 0
```

Comment:
this looks a bit messed up because I used an Either and Tuple2 rather than going full custom ADT. (Right is info needed for corporate, Left is the info needed for standard subs)
I realised the types were not specific enough when I was trying to unit test it, as I would have had to make up loads of irrelevant data.  Happy to go for a custom ADT or other solution to make things more clear in the pattern matching.